### ISSUE TITLE
[Bug] BoxBilling does not correctly identify country

### ISSUE THREAD

**Issue by BenNottelling** [boxbilling/boxbilling#945]
### What happened?

BoxBilling doesn't correctly identify that I live in the United States. Easy way to test is to enable tax and setup a tax rule for your country, you'll find that the tax doesn't show up when you go to purchase.
Set the tax rate to be for all countries and then it'll show up.

### BoxBilling Version

Current Master Branch

### What version of PHP are you running?

7.4x

### Additional Notes

I have tried updating geoip2 and the ```GeoLite2-Country.mmdb``` file but it made no difference.
These updates will still be included in my dependency update PR anyways since it's best practice to keep things up to date

**Comment by notbasetwo** [comment]
Where is the license key stored in BoxBilling for the GeoLite2 data?

**Comment by BenNottelling** [comment]
> Where is the license key stored in BoxBilling for the GeoLite2 data?

To the best of my knowledge, this shouldn't be needed to identify the country. Inside of `src\bb-library\` we have `GeoLite2-Country.mmdb` which is the DB used for it. If we want to pull it from their website rather than a local DB, I think that's when we need the license key.

I updated the DB and the accompanied library in this commit, but it made no difference https://github.com/boxbilling/boxbilling/commit/1e05b1d9354a8fe83f727435dc809d5b96f4cc45

**Comment by notbasetwo** [comment]
> 
> 
> > Where is the license key stored in BoxBilling for the GeoLite2 data?
> 
> To the best of my knowledge, this shouldn't be needed to identify the country. Inside of `src\bb-library\` we have `GeoLite2-Country.mmdb` which is the DB used for it. If we want to pull it from their website rather than a local DB, I think that's when we need the license key.
> 
> I updated the DB and the accompanied library in this commit, but it made no difference [1e05b1d](https://github.com/boxbilling/boxbilling/commit/1e05b1d9354a8fe83f727435dc809d5b96f4cc45)

You are not permitted to redistribute this database as per the GeoLite2 EULA.

**Comment by BenNottelling** [comment]
> You are not permitted to redistribute this database as per the GeoLite2 EULA, iirc.
> 
> Correction: I think you can, but I don't see the Maxmind disclaimer in the repo.

I just looked at the site and it looks like you're right. Previous contributors seem to have a history of integrating things without disclosing the license. Could you please add this disclaimer to your existing PR so we can get that taken care of quickly?

**Comment by notbasetwo** [comment]
> 
> 
> > You are not permitted to redistribute this database as per the GeoLite2 EULA, iirc.
> > Correction: I think you can, but I don't see the Maxmind disclaimer in the repo.
> 
> I just looked at the site and it looks like you're right. Previous contributors seem to have a history of integrating things without disclosing the license. Could you please add this disclaimer to your existing PR so we can get that taken care of quickly?

Done, good shout.

**Comment by notbasetwo** [comment]
I don't believe this is a bug, this should be changed to a feature request - and may be more helpful to open a discussion about how this should be implemented, whether GeoIP should be used or just asking the user what country they belong to.

The registration form is actually quite awful when purchasing a new product so perhaps the discussion needs to evolve beyond that.

The only place I can find the usage of GeoIP seems to be related to displaying IP addresses in the administration panel, which is working fine.

Willing to be proven wrong though, if someone can point me to where in code this is actually broken.

**Comment by BenNottelling** [comment]
@notbasetwo Oh, it looks like you are correct. I incorrectly assumed that that DB was used to determine the country since I wasn't required to enter it to make a purchase, or create an account.

Something should probably be changed with how all of that works, would make a good discussion

**Comment by notbasetwo** [comment]
It would be nice to use it to automatically point the country selection on registration to the right place, the "default country" approach used by a lot of other billing systems. But the user's current country may not necessarily be correct (or the GeoIP data could just be wrong) - so yeah, a very interesting discussion to be had here!

GeoLite2 data could be used well, but at the moment isn't being used for anything useful - I agree,

It could potentially be useful for something like a very basic anti-fraud feature, but that is something for another discussion.
